sdk/python/tests/integration/online_store/test_universal_online.py

```diff
@@ -1052,6 +1045,30 @@ def test_go_server_life_cycle(go_cycle_environment, go_data_sources):
             pass
     assert not go_server_still_alive
 
+    # Yield control to monitor thread to restart process.
+    time.sleep(1)
+
+    # Make sure the background thread has created a new subprocess.
+    assert (
+        fs._go_server
+        and fs._go_server._go_server_started.is_set()
+        and fs._go_server._shared_connection._process
+    )
+    new_go_fs_pid = fs._go_server._shared_connection._process.pid
+    assert new_go_fs_pid != go_fs_pid
+
+    # Ensure process is still running.
+    assert fs._go_server._shared_connection._process.poll() is None
+
+    # And we can still get feature values.
+    fs.get_online_features(features=features, entity_rows=entity_rows).to_dict()
+
+    fs.kill_go_server()
+    time.sleep(1)
+
+    # Ensure process is dead.
+    assert fs._go_server._shared_connection._process.poll() is not None
```

Comment:
Should we also figure out if the daemon thread is dead?

Reply:
yes